src/views/AppView.jsx

```diff
@@ -4,20 +4,18 @@ import { withRouter, Route, Redirect } from 'react-router-dom';
 import { NotificationContainer, NotificationManager } from 'react-notifications';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import HomeView from 'views/HomeView.jsx';
-import SignInView from 'views/SignInView.jsx';
-import SignUpView from 'views/SignUpView.jsx';
-import ImagesView from 'views/ImagesView.jsx';
-import AddImageView from 'views/AddImageView.jsx';
 import Header from 'components/Header/Header.jsx';
 import Main from 'components/Main/Main.jsx';
 import Footer from 'components/Footer/Footer.jsx';
 import { signOut, autoLogin } from 'actions/user.js';
+import routes from 'routes.js';
 
 class App extends Component {
 
   componentDidMount() {
-    this.props.autoLogin();
+    if (!this.props.isLoggedIn) {
+      this.props.autoLogin();
```

Comment:
Since we are hydrating the app in the server, we no longer need to call the API to check if the user is logged in